src/pages/_middleware.ts

```diff
@@ -0,0 +1,28 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { rewrites } from '../../config/rewrites';
+
+export async function middleware(req: NextRequest) {
+    const { pathname } = req.nextUrl;
+
+    // Rewrites are done in this middleware due to
+    // existing issue with SSG and dynamic routing
+    // when using built-in rewrites configuration.
```

Comment:
Is there an issue opened on the Next.js github repo for this? If so, could you send that link? It would be nice to track that.

Reply:
I believe it is this one on a canary release https://github.com/vercel/next.js/pull/32767. But there may be other issues specifically with the way we use SSG and dynamic routing that is unique and might require additional research and potentially opening another issue with Next.js.